core/src/main/java/org/infinispan/persistence/async/AsyncCacheWriter.java

```diff
@@ -201,14 +207,18 @@ public void write(MarshallableEntry entry) {
    }
 
    @Override
-   public void writeBatch(Publisher publisher) {
-      putAll(
-            Flowable.fromPublisher((Publisher<MarshallableEntry>) publisher)
-                  .map(me -> new Store(me.getKey(), me))
-                  .cast(Modification.class)
-                  .toList()
-                  .blockingGet()
-      );
+   public CompletionStage<Void> writeBatch(Publisher publisher) {
```

Comment:
No types for the argument?

Reply:
Unfortunately the `AsyncCacheWriter` has never had generics:

https://issues.jboss.org/browse/ISPN-7939